### PR TITLE
Thumbnails

### PR DESCRIPTION
**Pull request by ddvk**
generate thumbnails (book cover) when uploading pdfs.

the device does not generate thumbnails when the pdf size is larger than 20mb.

### CODE REVIEW

File: annotations/license.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"time"
+	_ "unsafe"
+
+	"github.com/unidoc/unipdf/v3/common/license"
+)
+
+//go:linkname licenseKey github.com/unidoc/unipdf/v3/common/license.licenseKey
+var licenseKey *license.LicenseKey
```
Comment (juruen) [inline annotations/license.go:11]: Oh, so that's how you are supposed to use the community license :)

Great to know!

File: util/zipdoc.go [hunk 2]
```diff
@@ -30,7 +61,10 @@ func CreateZipDocument(id, srcPath string) (string, error) {
 	}
 
 	tmp, err := ioutil.TempFile("", "rmapizip")
-	log.Trace.Println("creating temp zip file:", tmp.Name())
+	if err != nil {
+		return "", err
+	}
+	// log.Trace.Println("creating temp zip file:", tmp.Name())
```
Comment (juruen) [inline util/zipdoc.go:67]: Could we please remove this comment?

File: util/zipdoc.go [hunk 3]
```diff
@@ -83,7 +132,7 @@ func CreateZipDocument(id, srcPath string) (string, error) {
 
 func CreateZipDirectory(id string) (string, error) {
 	tmp, err := ioutil.TempFile("", "rmapizip")
-	log.Trace.Println("creating temp zip file:", tmp.Name())
+	// log.Trace.Println("creating temp zip file:", tmp.Name())
```
Comment (juruen) [inline util/zipdoc.go:135]: Same as above, but if you want to keep the trace logging is fine, unless it's broken you should be able to set the logging level through an env variable.

Comment (ddvk) [reply util/zipdoc.go:135]: the logging was not initialized in the unit test (which failed with nil ref), was not sure how to solve it

**Comment by ddvk** [comment]
> A couple of nits, but otherwise 👍 
> 
> 
> 
> And much ❤️ for showing me how to set the community license :)

that was just a dirty hack ;). 

i think you’re supposed to have a “real” license and pass it via env variable or set it in setlicens

**Comment by juruen** [comment]
> > A couple of nits, but otherwise 👍
> > And much ❤️ for showing me how to set the community license :)
> 
> that was just a dirty hack ;).
> 
> i think you’re supposed to have a “real” license and pass it via env variable or set it in setlicens

I tried to unsuccessfully contact them in multiple ways to see how that works. That's why I ended up forking the unipdf project and changing rmapi's license to AGPL.

File: util/zipdoc.go (hunk 3 above)
Comment (juruen) [reply util/zipdoc.go]: we'll need to add a null logger for tests at some point to do this right.

**Comment by juruen** [comment]
@ddvk feel free yo merge this PR at your convinience

**Pull request merged**